sender.go

```diff
@@ -97,6 +99,19 @@ func (sender *Sender) Backlog() (incomingRequests, incomingResponses, outgoingRe
 // Stops the sender's goroutine.
 func (sender *Sender) Stop() {
 	sender.queue.stop()
+
+	// There can be goroutines spawned by message.asyncRead() that are blocked waiting to
+	// read off their end of an io.Pipe, and if the peer abruptly closes a connection which causes
+	// the sender to stop(), the other side of that io.Pipe must be closed to avoid the goroutine's
+	// call to unblock on the read() call.  This loops through any io.Pipewriters in pendingResponses and
+	// close them, unblocking the readers and letting the message.asyncRead() goroutines proceed.
+	for _, msgStreamer := range sender.receiver.pendingResponses {
```

Comment:
I think this use of pendingResponses needs locking. Make this loop a method on `receiver`.
Since this operates on the receiver, should it be part of the receiver's teardown code, instead of being in sender.Stop?